docs/DataRequirements.md

```diff
@@ -0,0 +1,252 @@
+# Hermes IBC Relayer Data Requirements
+
+## Changelog
+
+- 2024-01-17: Initial requirements
+
+## Context
+
+The following endpoints (or equivalent) are necessary for operating the relayer.
+An optimal approach involves encapsulating these endpoints as methods on the
+unified clients designed to manage RPC or WebSocket requests and responses. For
+each section, we provide a list of the endpoints and the latest status of that
+endpoint, as far as we could investigate in the Sovereign SDK codebase. They are
+ordered from highest to lowest impact roughly, i.e., the last endpoint in the
+list is the least important and least frequently required.
+
+## Table of Contents
+
+- [Hermes IBC Relayer Data Requirements](#hermes-ibc-relayer-data-requirements)
+  - [Changelog](#changelog)
+  - [Context](#context)
+  - [Table of Contents](#table-of-contents)
+  - [Sequencer RPC](#sequencer-rpc)
+    - [`/send_transactions`](#send_transactions)
+    - [`/send_evidence`](#send_evidence)
+    - [`/sequencer_health`](#sequencer_health)
+  - [Rollup RPC](#rollup-rpc)
+    - [`/transaction_search`](#transaction_search)
+    - [`/aggregated_proof_search`](#aggregated_proof_search)
+    - [`/rollup_params`](#rollup_params)
+    - [`/status`](#status)
+    - [`/rollup_health`](#rollup_health)
+  - [IBC modules RPC](#ibc-modules-rpc)
+    - [Channel endpoints](#channel-endpoints)
+    - [Client endpoints](#client-endpoints)
+    - [Connection endpoints](#connection-endpoints)
+  - [Rollup WebSocket](#rollup-websocket)
+    - [`/subscribe_aggregated_proofs`](#subscribe_aggregated_proofs)
+    - [`/subscribe_events`](#subscribe_events)
+
+## Sequencer RPC
+
+### `/send_transactions`
```

Comment:
The `sequencer_acceptTx` RPC method on the sequencer does not seem to validate or return anything. This probably means there is no tx hash returned.

There is a separate `sequencer_publishBatch` method, which would submit all pending transactions as a blob to the DA. However it also only return a string that reports how many transactions have been published. This would probably be an internal method of the sequencer, as it doesn't make sense if any user can trigger the publishing at any time if the sequencer is exposed to public.

One possibility is that we can ask to get the tx hash of the blob submission to the Celestia DA. If we know which blob contains the transaction, we can probably trace it back to the rollup based on the blob hash.

Ideally though, all those should be keep tracked of on the rollup side. The sequencer needs to provide a tx hash for both the rollup tx and the blob tx, which can be traced back from the local rollup node of the relayer operator.

Reply:
Found the calculation of the tx hash to be done here: https://github.com/Sovereign-Labs/sovereign-sdk/blob/nightly/full-node/sov-sequencer/src/batch_builder.rs#L163-L167

However the sequencer only prints the hash to the log and not return anything. On the other hand, it might be possible for us to deterministically get the tx hash without the sequencer, since it is really a hash of the tx raw bytes.

On the other hand, here it shows that if there are failure in the tx at the time of blob inclusion, it would just get silently discarded: https://github.com/Sovereign-Labs/sovereign-sdk/blob/nightly/full-node/sov-sequencer/src/batch_builder.rs#L147-L150

We probably need the sequencer to have an API to poll or subscribe to the status of the pending transactions, until the blob is submitted (or ideally be committed) to the DA.